### PR TITLE
Obsv as options

### PR DESCRIPTION
**Pull request by thorfour**
Makes logging, tracing, and metrics options instead of required params.

### CODE REVIEW

File: query/engine.go [hunk 1]
```diff
@@ -25,16 +25,30 @@ type LocalEngine struct {
...
+type Option func(*LocalEngine)
+
+func WithTracer(tracer trace.Tracer) Option {
+	return func(e *LocalEngine) {
+		e.tracer = tracer
+	}
+}
+
 func NewEngine(
 	pool memory.Allocator,
```
Comment (metalmatze) [inline query/engine.go:37]: I always wonder if we should do the same with these `memory.Allocator`.
Totally out of scope for this PR, maybe creating an issue for outside contributions. That would be a nice one, I imagine.

**Review by asubiotto — APPROVED** [review]
Thank you!

**Review by metalmatze — APPROVED** [review]
Awsome. Thanks for cleaning this up!

File: query/engine.go (hunk 1 above)
Comment (thorfour) [reply query/engine.go:37]: Totally agree, I strongly considered removing it in this PR as well but decided to leave it for now

**Pull request merged**